driver-core/src/main/java/com/datastax/driver/core/RegularStatement.java

```diff
@@ -190,6 +189,47 @@ public boolean hasValues() {
     }
 
     /**
+     * Returns the keyspace this query operates on.
+     * <p/>
+     * Unless the keyspace has been explicitly set through {@link #setKeyspace},
+     * this method will return {@code null} to avoid having to parse the query
+     * string.
+     *
+     * @return the keyspace set through {@link #setKeyspace} if such keyspace was
+     * set, {@code null} otherwise.
+     * @see Statement#getKeyspace
+     */
+    @Override
+    public String getKeyspace() {
```

Comment:
Added default implementation to avoid clirr errors, we could also consider simply defining keyspace in `RegularStatement` since I think most implementations do something similar and we can handle the few cases that don't.